### PR TITLE
feat(observer): only hash the first 1MiB of data retrieved

### PR DESCRIPTION
**Pull request by karlprieb**
`getArnsResolution` will only hash the fist 1MiB of data instead of the whole data.

### CODE REVIEW

File: src/observer.ts [hunk 1]
```diff
@@ -64,20 +64,37 @@ function getArnsResolution({
   });
   const dataHash = crypto.createHash('sha256');
 
+  let streamBytesProcessed = 0;
+  const MAX_BYTES_TO_PROCESS = 1048576; // 1MiB
```
Comment (dtfiedler) [inline src/observer.ts:68]: nit: i'd say we could make this configurable but someone might try and set it to something ridiculously small so fine as a const

Comment (karlprieb) [reply src/observer.ts:68]: I don't think it should be configurable as all observers should hash the same data

File: src/observer.ts [hunk 2]
```diff
@@ -88,19 +105,27 @@ function getArnsResolution({
...
+        dataHash.update(data.slice(0, bytesToProcess));
+        streamBytesProcessed += bytesToProcess;
+      }
+
+      if (streamBytesProcessed >= MAX_BYTES_TO_PROCESS) {
+        stream.on('close', () => {
+          resolveWithResponse(response);
+        });
+
+        stream.destroy();
```
Comment (karlprieb) [inline src/observer.ts:123]: @dtfiedler @djwhitt I had to listen to `close` here as `stream.destroy()` will close the response so `end` will never be emitted.

I listen to this only inside this if so we don't resolve twice as `got` will emit `close` AND `end` when a request ends normally (without empirically destroying it).

Is there a better way to handle this?

Comment (djwhitt) [reply src/observer.ts:123]: Nah, this seems fine to me. You could make the event handler more clever instead, but I think that just obscures what's actually happening.

**Review by djwhitt — APPROVED** [review]

**Pull request merged**